libs/hdf-converters/src/veracode-mapper.ts

```diff
@@ -0,0 +1,552 @@
+/* eslint-disable prettier/prettier */
+import { integer } from 'aws-sdk/clients/cloudfront';
+import parser from 'fast-xml-parser';
+import {ExecJSON} from 'inspecjs';
+import _ from 'lodash';
+import {version as HeimdallToolsVersion} from '../package.json';
+import {
+  BaseConverter,
+  ILookupPath,
+  MappedTransform
+} from './base-converter';
+import {CweNistMapping} from './mappings/CweNistMapping';
+const CWE_LENGTH = 'cwe.length'
+const SCA_VULNERABILITIES = 'detailedreport.software_composition_analysis.vulnerabilities['
+const REPORT_SEVERITY = 'detailedreport.severity['
+const SCA_CVES = 'detailedreport.software_composition_analysis.cves['
+const CATEGORY = '].category['
+const CWE_NIST_MAPPING = new CweNistMapping();
+const DEFAULT_NIST_TAG = ['SI-2', 'RA-5'];
+const IMPACT_MAPPING: Map<string, number> = new Map([
+  ['5', 0.9],
+  ['4', 0.7],
+  ['3', 0.5],
+  ['2', 0.3],
+  ['1', 0.1],
+  ['0', 0.0]
+]);
+
+function impactMapping(severity: number): number {
+  if (typeof severity === 'string' || typeof severity === 'number') {
+    return IMPACT_MAPPING.get(severity.toString().toLowerCase()) || 0.1;
+  } else {
+    return 0.1;
+  }
+}
+
+function nistTag(input: string): string[] {
+  const len = Number(`${_.get(input, CWE_LENGTH)}`)
+  const cwes = []
+  for (let index = 0; index < len; index++) {
+    cwes.push(`${_.get(input, 'cwe.cweid')}`)
+  }
+  return CWE_NIST_MAPPING.nistFilter(cwes, DEFAULT_NIST_TAG).concat(['Rev_4']);
+}
+
+function formatRecommendations(input: unknown): string {
+  const text = [];
+  if (_.has(input, 'recommendations.para')) {
```

Comment:
my understanding from reading this code is that recommendations.para could be a single object or an array of those objects from which we need to extract this text attribute

i would replace the inside of this if block with something like the following:

```
if(Array.isArray(_.get(input, 'recomenndations.para'))) {
that same map trick from above but maybe not using destructuring unless you can guarantee that each one of these para objects contains 'text' which i think we can't cause the non-array variant needs to check if it's there
} else {
what you got there but with lodash has against recommendations.para.text instead of doing a get and comparing it to undefined
}
```